app/controllers/api/v3/imo_callbacks_controller.rb

```diff
@@ -0,0 +1,35 @@
+class Api::V3::ImoCallbacksController < ApplicationController
+  skip_before_action :verify_authenticity_token
+  before_action :authenticate_request
+
+  class ImoCallbackError < StandardError; end
+
+  rescue_from ImoCallbackError do
+    head :bad_request
+  end
+
+  def subscribe
+    unless permitted_params[:event] == "accept_invite"
+      raise ImoCallbackError.new("unexcepted Imo invitation event: #{permitted_params[:event]}")
+    end
+
+    patient = Patient.find(permitted_params[:patient_id])
+    unless patient.imo_authorization
+      raise ImoCallbackError.new("patient #{patient.id} does not have an ImoAuthorization")
+    end
+    patient.imo_authorization.status_subscribed!
+    head :ok
+  end
+
+  private
+
+  def permitted_params
+    params.permit(:patient_id, :event)
+  end
+
+  def authenticate_request
+    authenticate_or_request_with_http_basic do |username, password|
```

Comment:
Looks like it can be even simpler - see https://api.rubyonrails.org/v5.2.1/classes/ActionController/HttpAuthentication/Basic.html

```
   http_basic_authenticate_with name: "dhh", password: "secret", except: :index
```